pkg/pillar/cmd/zedrouter/pubsubhandlers.go

```diff
@@ -122,11 +122,50 @@ func (z *zedrouter) handleDNSImpl(ctxArg interface{}, key string,
 			z.log.Errorf("handleDNSImpl: failed to get config for NI %s", niStatus.UUID)
 			continue
 		}
-		z.doUpdateNIUplink(niStatus.SelectedUplinkLogicalLabel, &niStatus, *niConfig)
+		// Handle change of the configured port label.
+		niStatus.PortErr.ClearError()
+		changedPorts, portErr := z.updateNIPorts(&niStatus)
+		if portErr != nil {
+			portErr = fmt.Errorf(
+				"failed to update selection of ports for network instance %s: %v",
+				niStatus.UUID, portErr)
+			z.log.Error(portErr)
+			niStatus.PortErr.SetErrorNow(portErr.Error())
+		}
+
+		if changedPorts {
+			// Changing the number of ports may affect if a multi-path default route
+			// is needed or not.
+			_ = z.updateNIRoutes(&niStatus, false)
+		}
+
+		// Re-check MTUs between the NI and the selected ports.
+		mtuToUse, mtuErr := z.checkNetworkInstanceMTUConflicts(*niConfig, &niStatus)
+		niStatus.MTU = mtuToUse
+		if mtuErr == nil && niStatus.MTUConflictErr.HasError() {
+			// MTU conflict was resolved.
+			niStatus.MTUConflictErr.ClearError()
+		}
+		if mtuErr != nil &&
+			mtuErr.Error() != niStatus.MTUConflictErr.Error {
+			// New MTU conflict arose or the error has changed.
+			z.log.Error(mtuErr)
+			niStatus.MTUConflictErr.SetErrorNow(mtuErr.Error())
+		}
+
+		// Apply port/MTU changes in the network stack.
+		if niStatus.Activated {
+			z.doUpdateActivatedNetworkInstance(*niConfig, &niStatus)
+		}
+		if niConfig.Activate && !niStatus.Activated && niStatus.EligibleForActivate() {
+			z.doActivateNetworkInstance(*niConfig, &niStatus)
+			z.checkAndRecreateAppNetworks(niStatus.UUID)
+		}
+		z.publishNetworkInstanceStatus(&niStatus)
 	}
 
-	if z.uplinkProber != nil {
-		z.uplinkProber.ApplyDNSUpdate(status)
+	if z.portProber != nil {
```

Comment:
wouldn't this z.portProber already exist always if we receive the DNS?

Reply:
Good point, I have removed the unnecessary check.